lib/index.js

```diff
@@ -45,26 +45,29 @@ bedrock.events.on('bedrock-mongodb.ready', function init(callback) {
  *
  * @param actor the Identity performing the action.
  * @param id the ID of the identity to check.
+ * @param [options] the options touse.
+            [deleted] true to check identities marked as deleted.
  * @param callback(err, exists) called once the operation completes.
  */
-api.exists = function(actor, id, callback) {
-  async.auto({
-    checkPermission: callback => {
-      api.checkPermission(
-        actor, PERMISSIONS.IDENTITY_ACCESS, {resource: id}, callback);
-    },
-    checkExists: ['checkPermission', callback => {
-      database.collections.identity.find({
-        id: database.hash(id)
-      }, {
-        limit: 1
-      }).count((err, count)=> {
-        callback(err, !!count);
-      });
-    }]
-  }, (err, results) => {
-    callback(err, results.checkExists);
-  });
+api.exists = function(actor, id, options, callback) {
+  if(typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  options = options || {};
+  async.series({
```

Comment:
That's fine.  If this is in a critical path, and it's assumed checkPermission will usually pass, then it might be faster to do operations in parallel.  The low level calls do some async DB I/O so starting the second call while the first is executing could save time.  We can revisit if we do some profiling and find this is a hot spot.